### ISSUE TITLE
Custom properties should support :host(selector) rules

### ISSUE THREAD

**Issue by sorvell** [Polymer/polymer#1787]
Currently properties cannot be defined or consumed within `:host` rules that have a selector in the css function argument (e.g. `:host(selector)`).

We can and should be able to add this feature. This is a tracking issue for the feature.

**Comment by ebidel** [comment]
Any progress on this? 

For I/O, we have rules like this that no longer work under shady dom:

```
:host(.bg-dark-grey, .bg-photo) #o2 {
  background-color: ...
}
```

**Comment by TimvdLippe** [comment]
As a temporary workaround, you can use `host-context` (example: http://codepen.io/robdodson/pen/ftpoG/)

**Comment by sorvell** [comment]
This looks like a parsing bug with including multiple simple selectors in the `:host` function. As a workaround, you can comma separate the entire selector, `:host(.bg-dark-grey) #o2, :host(.bg-photo) #o2` (e.g. http://jsbin.com/yoraji/edit?html,output)

**Comment by azakus** [comment]
Fixed with #3587
